### ISSUE TITLE
Shift-drag behaviourisms

### ISSUE THREAD

**Issue by almereyda** [fedwiki/wiki-client#161]
The shift-drag is intended well, but its current implementation retains few glitches:

* The source item from the copy changes place in DOM. Copy is not obvious up until a full page refresh.
* Shift-dragging onto identical items could help transclude the conversation even more. (Don't feel capable formulating this from the 'what is lacking' perspective.)

Were these accepted as those and documented before?
Else both could become individual issues and this closed.

**Comment by almereyda** [comment]
Easing the shift-drag-replacement animation could help to avoid clutter.

**Comment by WardCunningham** [comment]
Yes, this drag logic using jquery-ui sortable is old and insufficient to intent. There are examples in the jquery-ui documentation that do most things right. I've studied these but not found the combination that works. https://github.com/fedwiki/wiki-client/blob/master/lib/refresh.coffee#L49-L115

A good model is the way the mac finder column browser works, especially how it distinguishes copy from move by showing a green plus sign over the item being moved. Shift here reverses the decision that the browser has already made. This idiom dates back to Xanadu. https://en.wikipedia.org/wiki/Miller_columns

**Comment by paul90** [comment]
If I remember correctly, there are a number of assumptions that made in jQuery-ui sortable that make it difficult, if not impossible, to mix the drag to move and shift-drag to copy. The only jQuery drag a copy examples I have seen use draggable.

This is very probably an area where a native HTML5 solution could well be easiest [HTML Drag Operations](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Drag_operations) - but are likely to require some extensive changes to the client code. Couple this with other desirable client changes this will likely be easiest to do as part of a rewrite.

**Comment by WardCunningham** [comment]
@paul90 Thanks for pointing this documentation out. We could learn a lot from some use-case specific and federation-compatible rewrites. A read-only version or a touch-specific version come to mind. The addition of one extra layer of div (.paper) was fraught with complications. I suspect redoing drag/drop would be equal, but maybe no worse.
